packages/spacebars-tests/template_tests.js

```diff
@@ -3067,3 +3067,38 @@ Tinytest.add("spacebars-tests - template_tests - inclusion with data remove (#31
   test.isTrue(parentView.isDestroyed);
   test.equal(canonicalizeHtml(div.innerHTML), "<span></span>");
 });
+
+testAsyncMulti("spacebars-tests - template_tests - template-level subscriptions", [
+  function (test, expect) {
+    var tmpl = Template.spacebars_template_test_template_level_subscriptions;
+    var tmplInstance;
+
+    // Make sure the subscription stops when the template is destroyed
+    var stopCallback = expect();
+
+    // Make sure the HTML is what we expect when the subscription is ready
+    var checkHTMLThenRemove = expect(function () {
+      test.equal(canonicalizeHtml(div.innerHTML), "ready! true");
+
+      // This will call the template's destroyed callback
+      Blaze.remove(tmplInstance.view);
+    });
+
+    // Manually check the subscribe ready callback to make sure the template is
+    // doing the right thing
+    var subscribeCallback = expect(function () {
+      Tracker.afterFlush(checkHTMLThenRemove);
+    });
+
+    tmpl.onCreated(function () {
+      subHandle = this.subscribe("items", subscribeCallback);
```

Comment:
Maybe have more than one subscription? So that you test the behavior of "they all get stopped", and the behavior of "only if they are all ready is `Template.ready` true"